RMB-724, RMB-718: totalRecords=exact/estimated/none/auto

julianladisch opened the pull request:
https://issues.folio.org/browse/RMB-724
https://issues.folio.org/browse/RMB-718

julianladisch committed:
RMB-724, RMB-718: totalRecords=exact/estimated/none/auto

https://issues.folio.org/browse/RMB-724
https://issues.folio.org/browse/RMB-718

adamdickmeiss reviewed (approved):
OK. The enum will be a per-model type .. eg BeesHistoryGetTotalRecords in RMB.. The problem with that IMHO is that the model (BeesHistoryGetTotalRecords) is not really appropriate type for this it is used everywhere. A string would be better, but that has the drawback of not specifying value values.

julianladisch committed:
Use string, not enum, for easy calling of PgUtil methods

julianladisch committed:
Move totalRecords before offset

Another PR removes the `String lang` parameter:

If we insert totalRecords after limit then the method parameter list of a generated get method
`String query, int offset, int limit, String lang, Map<String, String> okapiHeaders, ...`
will become
`String query, int offset, int limit, String totalRecords, Map<String, String> okapiHeaders, ...`

This goes unnoticed.

Moving totalRecords before offset yields
`String query, String totalRecords, int offset, int limit, Map<String, String> okapiHeaders, ...`
so that the method has a different signature, making developers aware of this
change and nuging them to pass totalRecords to query methods.

julianladisch committed:
Merge branch 'raml1.0' into RMB-724-RMB-718-totalRecords-none

julianladisch commented:
Good catch, the multiple enum Java classes will make it difficult to invoke the PgUtil.get method. Therefore I changed it from enum to String.
